test/Dex.test.js

```diff
@@ -15,6 +17,169 @@ contract("IntegratedLimitOrderDex", (accounts) => {
   });
 
   it("should support placing sell orders", async () => {
-    await dex.sell(100, 10);
+    const oldBalance = (await dex.balanceOf.call(accounts[0]))
+    let tx = await dex.sell(100, 10);
+    const newBalance = (await dex.balanceOf.call(accounts[0]))
+
+    assert.equal(newBalance, oldBalance - 100, "Sell order did not update balance")
+    assert.equal(tx.logs[2].event, "NewSellOrder", "Incorrect event")
+    assert.equal(tx.logs[3].event, "OrderIncrease", "Incorrect event")
+  });
+  
+  it("should support placing multiple buy orders", async () => {
+    const amount = 1;
+    const oneEtherInWei = web3.utils.toWei("1");
+    const price = web3.utils.toBN(oneEtherInWei);
+    const numBuys = 5;
+    
+    let tx;
+    for (let i = 0; i < numBuys; i++) {
+      tx = await dex.buy(amount, price, {value: oneEtherInWei});
+    }
+
+    let orderQuantity = (await dex.getOrderQuantity(price))
+    assert.equal(orderQuantity, numBuys, "Buy order did not update balance")
+    assert.equal(tx.logs[0].event, "OrderIncrease", "Unexpected event emitted")
+  });
+  
+  it("should support fully filling sell order", async () => {
+    const oneEtherInWei = web3.utils.toWei("1");
+    const price = web3.utils.toBN(oneEtherInWei);
+    const amount = 5;
+    
+    let tx = await dex.sell(amount, price);
+    
+    let orderQuantity = (await dex.getOrderQuantity(price))
+    let balance = (await dex.getEthBalance(accounts[0]))
+    assert.equal(orderQuantity, 0, "Order quantity did not update balance")
+    assert.equal(balance, price * 5, "Ethereum balance did not update")
```

Comment:
Also needs to check the ERC20 transfers occurred for the sold/purchased tokens. Using a second account to place the buy orders would likely be optimal in order to separate the two.

Reply:
Good idea, ill add that